remote/conftest.py

```diff
@@ -0,0 +1,133 @@
+# coding=utf-8
+"""Pytest customizations and fixtures for tests to execute on remote hosts."""
+import os
+import pytest
+from camayoc import utils, config
+from time import sleep
+from remote.constants import (
+    DATA_CENTER,
+    CLUSTER,
+    HOST,
+)
+
+
+@pytest.fixture
+def isolated_filesystem():
+    """Fixture that creates a temporary directory.
+
+    Changes the current working directory to the created temporary directory
+    for isolated filesystem tests.
+    """
+    with utils.isolated_filesystem() as path:
+        yield path
+
+
+@pytest.fixture(scope='session', autouse=True)
+def manage_systems():
+    """Fixture that manages remote systems.
+
+    At the beginning of a session, hosts will be turned on. We must wait for
+    the machines to complete booting so there is about 30 seconds of overhead
+    at the beginning before test begin executing.
+
+    At the end of the session, they are turned off.
+
+    Hosts must have a "provider" specified in the config file, otherwise they
+    will be skipped and assumed to be accessible by rho.
+
+    Hosts that are powered on at the beginning of the session are then powered
+    down at the end of the session.
+
+    Example::
+
+        #/home/user/.config/camayoc/config.yaml
+        rho:
+          auths:
+            - username: root
+              name: sonar
+              sshkeyfile: /home/elijah/.ssh/id_sonar_jenkins_rsa
+          hosts:
+            - hostname: string-name-of-vm-on-vcenter
+              ip: XX.XX.XXX.XX
+              provider: vcenter
+              facts:
+                connection.port: 22
+                cpu.count: 1
+          profiles:
+            - name: profile1
+              auths:
+                - auth1
+              hosts:
+                - XX.XX.XXX.XX
+         vcenter:
+          hostname: **Required** -- url to vcenter instance
+          username: **optional** -- will first try environment variable $VCUSER
+          password: **optional** -- will first try environment variable $VCPASS
+
+    *Warning*
+    It is a bad idea to have multiple sessions of pytest running
+    concurrently against the same hosts/config file.
+    If one session ends before another, it will powerdown the
+    machines leaving the other one to fail.
+    """
+    from pyVim.connect import SmartConnect, Disconnect
+    import ssl
+
+    cfg = config.get_config()
+    s = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
+    s.verify_mode = ssl.CERT_NONE
+    # try and make a secure connection first
+    if 'vcenter' in cfg.keys():
```

Comment:
If vcenter is not available what happens?

Reply:
Do we want to continue the setup using env vars when there is no vcenter on the config file? Like everything is setup using env vars.